kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/zookeeper/ZookeeperDiscoveryClient.scala

```diff
@@ -290,6 +296,11 @@ class ZookeeperDiscoveryClient(conf: KyuubiConf) extends DiscoveryClient {
     secretNode.start()
   }
 
+  def getAndIncrement(path: String): Int = {
+    val dai = new DistributedAtomicInteger(zkClient, path, new RetryForever(1000))
+    dai.increment().preValue().intValue()
```

Comment:
from class `DistributedAtomicInteger` Java API documentation:

> The various increment methods return an [AtomicValue](https://curator.apache.org/apidocs/org/apache/curator/framework/recipes/atomic/AtomicValue.html) object. You must always check [AtomicValue.succeeded()](https://curator.apache.org/apidocs/org/apache/curator/framework/recipes/atomic/AtomicValue.html#succeeded()). None of the methods (other than get()) are guaranteed to succeed.

we need to retry `dai.increment()`